Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -0,0 +1,443 @@
+
+        /// <summary>
+        /// Gets or sets the Carousel orientation. Horizontal or Vertical
+        /// </summary>
+        public Orientation Orientation
+        {
+            get { return (Orientation)GetValue(OrientationProperty); }
+            set { SetValue(OrientationProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for MaxViewableItems.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty OrientationProperty = DependencyProperty.Register("Orientation", typeof(Orientation), typeof(Carousel), new PropertyMetadata(Orientation.Horizontal, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Each time a property change, we have to reposition
+        /// </summary>
+        private static void OnCarouselPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (e.NewValue == null)
+            {
+                return;
+            }
+
+            Carousel carouselControl = (Carousel)d;
+
+            if (e.Property == SelectedIndexProperty)
+            {
+                var item = carouselControl.Items[(int)e.NewValue];
+
+                // double check
+                if (carouselControl.SelectedItem != item)
+                {
+                    carouselControl.SetValue(SelectedItemProperty, item);
+                    return;
+                }
+            }
+
+            if (e.Property == SelectedItemProperty)
+            {
+                var index = carouselControl.Items.IndexOf(e.NewValue);
+
+                // double check
+                if (carouselControl.SelectedIndex != index)
+                {
+                    carouselControl.SetValue(SelectedIndexProperty, index);
+                    return;
+                }
+            }
+
+            carouselControl.UpdatePositions();
+        }
+
+        /// <summary>
+        /// Return ItemsPanel
+        /// </summary>
+        /// <returns>Gets the ItemsPanel used for the carousel</returns>
+        internal CarouselPanel GetItemsPanel()
+        {
+            return this.FindDescendant<CarouselPanel>();
+        }
+
+        /// <summary>
+        /// Returns the container used for each item
+        /// </summary>
+        /// <returns>Returns always a ContentControl</returns>
+        protected override DependencyObject GetContainerForItemOverride()
+        {
+            return new ContentControl();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Carousel"/> class.
+        /// </summary>
+        public Carousel()
+        {
+            // Set style
+            DefaultStyleKey = typeof(Carousel);
+            ManipulationMode = ManipulationModes.All;
```

Comment:
By the way, you solution make possible to remove the CarouselPanel reference in the Carousel control, smart ! :)